Stripe/STPShippingAddressViewController.m

```diff
@@ -0,0 +1,349 @@
+@property(nonatomic)NSString *currency;
+@property(nonatomic)STPTheme *theme;
+@property(nonatomic)PKShippingMethod *selectedShippingMethod;
+@property(nonatomic, weak)UITableView *tableView;
+@property(nonatomic, weak)UIImageView *imageView;
+@property(nonatomic)UIBarButtonItem *nextItem;
+@property(nonatomic)UIBarButtonItem *backItem;
+@property(nonatomic)UIBarButtonItem *cancelItem;
+@property(nonatomic)BOOL loading;
+@property(nonatomic)STPPaymentActivityIndicatorView *activityIndicator;
+@property(nonatomic)STPAddressViewModel *addressViewModel;
+@end
+
+@implementation STPShippingAddressViewController
+
+- (instancetype)init {
+    return [self initWithConfiguration:[STPPaymentConfiguration sharedConfiguration] theme:[STPTheme defaultTheme] currency:nil shippingAddress:nil selectedShippingMethod:nil];
+}
+
+- (instancetype)initWithConfiguration:(STPPaymentConfiguration *)configuration
+                                theme:(STPTheme *)theme
+                             currency:(NSString *)currency
+                      shippingAddress:(STPAddress *)shippingAddress
+               selectedShippingMethod:(PKShippingMethod *)selectedShippingMethod {
+    self = [super initWithNibName:nil bundle:nil];
+    if (self) {
+        _configuration = configuration;
+        _currency = currency ?: @"usd";
+        _theme = theme;
+        _selectedShippingMethod = selectedShippingMethod;
+        _addressViewModel = [[STPAddressViewModel alloc] initWithRequiredShippingFields:configuration.requiredShippingAddressFields];
+        _addressViewModel.delegate = self;
+        _addressViewModel.address = shippingAddress;
+        self.title = [self titleForShippingType:self.configuration.shippingType];
+    }
+    return self;
+}
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+    self.automaticallyAdjustsScrollViewInsets = NO;
+    UITableView *tableView = [[UITableView alloc] initWithFrame:CGRectZero style:UITableViewStyleGrouped];
+    tableView.sectionHeaderHeight = 30;
+    [self.view addSubview:tableView];
+    self.tableView = tableView;
+    self.backItem = [UIBarButtonItem stp_backButtonItemWithTitle:STPLocalizedString(@"Back", @"Text for back button")
+                                                           style:UIBarButtonItemStylePlain
+                                                          target:self
+                                                          action:@selector(cancel:)];
+    self.cancelItem = [[UIBarButtonItem alloc] initWithBarButtonSystemItem:UIBarButtonSystemItemCancel
+                                                                    target:self
+                                                                    action:@selector(cancel:)];
+    UIBarButtonItem *nextItem;
+    switch (self.configuration.shippingType) {
+        case STPShippingTypeShipping:
+            nextItem = [[UIBarButtonItem alloc] initWithTitle:STPLocalizedString(@"Next", nil)
```

Comment:
Don't add new `STPLocalizedString`s without developer comments (although FauxPas does not correctly catch them yet :\ )